File: app/src/main/java/com/grohden/niceanimals/ui/activities/GalleryActivity.java
package com.grohden.niceanimals.ui.activities;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.PagerSnapHelper;
import android.support.v7.widget.RecyclerView;

import com.grohden.niceanimals.R;
import com.grohden.niceanimals.realm.entities.NiceAnimal;
import com.grohden.niceanimals.ui.adapters.GalleryAdapter;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.realm.Realm;
import io.realm.RealmResults;

public class GalleryActivity extends AppCompatActivity {
    public static final String IMAGE_URL_EXTRA = "IMAGE_URL_EXTRA";

    @BindView(R.id.full_image_view)
    RecyclerView mNiceImageRV;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fullscreen_image);
        ButterKnife.bind(this);

        int imagePosition = getIntent().getIntExtra(IMAGE_URL_EXTRA, 0);
        configureRV(imagePosition);
    }

    private void configureRV(int position) {
        final RealmResults<NiceAnimal> niceAnimals = Realm.getDefaultInstance()
                .where(NiceAnimal.class)
                .findAll();
        final LinearLayoutManager manager = new LinearLayoutManager(
                this,
                LinearLayoutManager.HORIZONTAL,
                false
        );
        final GalleryAdapter galleryAdapter = new GalleryAdapter(niceAnimals);

        new PagerSnapHelper()
                .attachToRecyclerView(mNiceImageRV);

        mNiceImageRV.setLayoutManager(manager);
        mNiceImageRV.setAdapter(galleryAdapter);
        mNiceImageRV.scrollToPosition(position);
    }
}

File: app/src/main/java/com/grohden/niceanimals/ui/activities/GalleryActivity.kt
package com.grohden.niceanimals.ui.activities

import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.PagerSnapHelper
import blade.Blade
import blade.Extra
import com.grohden.niceanimals.R
import com.grohden.niceanimals.realm.entities.NiceAnimal
import com.grohden.niceanimals.realm.entities.NiceAnimalFields
import com.grohden.niceanimals.shibe.service.AnimalType
import com.grohden.niceanimals.ui.adapters.GalleryAdapter
import com.grohden.niceanimals.ui.base.BaseActivity
import com.grohden.niceanimals.ui.extensions.isEnum
import io.realm.Realm
import kotlinx.android.synthetic.main.gallery.*

@Blade
open class GalleryActivity : BaseActivity() {

    @Extra
    @JvmField
    var imagePosition: Int = 0

    @Extra
    lateinit var animalType: AnimalType

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.gallery)

        configureRV(imagePosition, animalType)
    }

    private fun configureRV(position: Int, animalType: AnimalType) {
        val niceAnimals = Realm.getDefaultInstance()
            .where<NiceAnimal>(NiceAnimal::class.java)
            .isEnum(NiceAnimalFields.TYPE, animalType)
            .findAll()

        val manager = LinearLayoutManager(
            this,
            LinearLayoutManager.HORIZONTAL,
            false
        )
        val galleryAdapter = GalleryAdapter(niceAnimals)

        PagerSnapHelper()
            .attachToRecyclerView(galleryRV)

        galleryRV.apply {
            layoutManager = manager
            adapter = galleryAdapter
            scrollToPosition(position)
        }
    }
}
